### PR TITLE
[SAAS-11757] Add Danger [WIP]

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
Adds DangerFile
https://danger.systems/ruby/

### CODE REVIEW

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please provide a summary in the PR description to make it easier to review</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR does not contain any JIRA issue keys in the PR title or commit messages (e.g. KEY-123)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dimagimon** [comment]
[33% (0.0%) vs master 33%](https://jenkins.dimagi.com/job/commcare-android-pr-job-builder/462298/)

**Comment by ShivamPokhriyal** [comment]
> 3 Warnings
> ⚠️	PR is classed as Work in Progress
> ⚠️	Please provide a summary in the PR description to make it easier to review
> ⚠️	This PR does not contain any JIRA issue keys in the PR title or commit messages (e.g. KEY-123)
> Generated by 🚫 [Danger](http://danger.systems/)

@shubham1g5 How does it look?

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

File: app/src/org/commcare/activities/CommCareActivity.java [hunk 1]
```diff
@@ -125,8 +125,10 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         FragmentManager fm = this.getSupportFragmentManager();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.GINGERBREAD_MR1) {
```
Comment (ShivamPokhriyal) [inline app/src/org/commcare/activities/CommCareActivity.java:128]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dimagi/commcare-android/blob/8eba354c593b70275ff43733ae020a7bc8ce7196/app/src/org/commcare/activities/CommCareActivity.java#L128"></span>Unnecessary; SDK_INT is always &gt;= 16</td>
    </tr>
  </tbody>
</table>

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+source 'https://rubygems.org'
+
+gem 'danger'
```
Comment (ShivamPokhriyal) [inline Gemfile:3]: I couldn't configure `bundle` with jenkins so this is not needed anymore.

File: Dangerfile [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+end
+
+# Ensure that labels have been used on the PR
+if github.pr_labels.empty?
+    failure "Please add labels to this PR"
+end
+
+# Make danger out jira ticket link
+jira.check(
+  key: ["SAAS"],
```
Comment (shubham1g5) [inline Dangerfile:18]: what does the key do here ?

**Review by shubham1g5 — APPROVED** [review]

File: Dangerfile (hunk 3 above)
Comment (ShivamPokhriyal) [reply Dangerfile:18]: The way it'll work is, you put the key in PR title or body and danger will produce the link in the comment.  For instance I put [SAAS-11757] in the title and danger commented the ticket link. 
And if we go further and integrate jira with github, then jira will use these keys from PR titles and add the PR/branch to the ticket automatically.

**Comment by ShivamPokhriyal** [comment]
Closed in favour of #2457

**Pull request closed without merging**